react/week2/my_first_app/src/index.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+//import App from './App';
+import * as serviceWorker from './serviceWorker';
+
+const root = document.getElementById('root');
+
+const todosArray = [
+  {
+    "id": 1,
+    "description": "Get out of bed"
+  },
+  {
+    "id": 2,
+    "description": "Brush teeth"
+  },
+  {
+    "id": 3,
+    "description": "Eat breakfast"
+  }, 
+  {
+    "id": 4,
+    "description": "Understand State and props"
+  },
+  {
+    "id": 5,
+    "description": "Passing the state through props"
+  }
+];
+class Header extends React.Component {
+  render() {
+    return <h1>Todolist</h1>
+  }
+}
+//I cannot pass a props that is in a state in one component and
+//set that props again in state in different component
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isChecked: false,
+      checked: []
+    };
+  }
+
+  handleCheckboxChange = (event) => {
+    const temp = event.target.checked
+    //Eventhough the item.id is int when it is passed inside name attribute in input tag
+    // it changes to string thats why parseInt
+    const specificTodo = parseInt(event.target.name)
+    //When I check the box my temp becomes true and executes if part
+    if (temp) {
+      //When checked I push that particular id to my checked array and update the state
+      this.state.checked.push(specificTodo)
+      this.setState({
+        isChecked: temp,
+        checked: this.state.checked
+      })
+    } else {
+      //When I uncheck, I get the index of the unchecked todo and remove it from checked array and update the state
+      const index = this.state.checked.indexOf(specificTodo)
+      this.state.checked.splice(index, 1);
+      this.setState({
+        isChecked: temp,
+        checked: this.state.checked
+      })
+    }
+  }
+
+  render() {
+    const array = this.props.todos;
+    const result = array.map((item) => {
+      return (
+        <div className='todo-list-display' key={item.id}>
+          <li style={{ 'textDecoration': this.state.checked.includes(item.id) ? 'line-through' : 'none' }}>{item.description}</li>
+          <label>
+            <input type="checkbox" name={item.id} onChange={this.handleCheckboxChange} defaultChecked={this.state.isChecked}></input>
```

Comment:
It doesn't look right to me that you are using a single variable to control the 'checked' state of all items. How does it know that an item is checked or not if you're using the same value for all items?

Reply:
Since its inside a map "this.state.checked.includes(item.id)" is going to be different for every item rendered in the list. Since the item.id is unique I am checking if that is available in my created checked array where I push this id. If a particular id is present it means the item is checked and I would get a strike through. I checked all the cases and it looks like its working as expected. Where am I missing??